arviz/plots/densityplot.py

```diff
@@ -206,23 +210,25 @@ def _d_helper(
     ax : matplotlib axes
     """
     if vec.dtype.kind == "f":
-        density, lower, upper = _fast_kde(vec, bw=bw)
-        x = np.linspace(lower, upper, len(density))
-        hpd_ = hpd(vec, credible_interval)
-        cut = (x >= hpd_[0]) & (x <= hpd_[1])
+        if credible_interval < 0.999:
```

Comment:
Could the check be `credible_interval != 1` or is it important that it truly is less than `.999`?

Reply:
This seem a better choice.